### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by scytacki**
Updates: 
- typescript
- react (to version 17)
- styled-components
- storybook
- create-react-app

The library was also tested locally with React 18. It seems to work fine so I updated the peerDependency spec to allow React 18

A few changes were needed to support version 6 of styled-components:
- the typing is more strict so a new `as` prop is used apply the same style to multiple SVG elements
- unknown props are now passed through by default to standard html elements. The exception are props that start with `$`. When tested with React 18 this issue became visible. 

I updated storybook, by removing all dependencies from package.json and then running `npx storybook@latest init` to add it back to the repo. Then I had to fix the config files. `storybook` complains about how we aren't using a create-react-app-preset addon. Supposedly the init command should have figured out we are using create-react-app and added automatically, but it didn't. I think the issue is that we haven't ejected, but we have customized various things. 
I tried adding the preset, but that broke various things. So I've just left it as is.

### CODE REVIEW

**Comment by scytacki** [comment]
The netlify checks are not configured in this repo as far as I can tell. It seems like the netlify deploy that Noah setup is monitoring this repo and trying to deploy PRs. I'd like to move on to Rete so I'm going to ignore them for now.

File: package.json [hunk 1]
```diff
@@ -34,34 +34,37 @@
...
   },
   "dependencies": {
-    "styled-components": "^5.2.3"
+    "styled-components": "^6.1.8"
   },
   "peerDependencies": {
-    "react": ">=16 <=17",
-    "react-dom": ">=16 <=17"
+    "react": ">=16 <=18",
+    "react-dom": ">=16 <=18"
```
Comment (kswenson) [inline package.json:41]: You could consider making it simply `>=16` and then we'd only have to change it if it breaks in a future version of React. 🤷

**Review by kswenson — APPROVED** [review]
👍 Looks good -- `create-react-app` is dead at this point and this project isn't even an app, so I think it'd be safe to eject at this point, but that can be for another day.

File: package.json (hunk 1 above)
Comment (scytacki) [reply package.json:41]: I kind of like the idea that we have to approve new versions of React. We can't really revise already released library versions, so if we say it supports everything 16 or higher we'd have a "lying" module sitting around.

**Pull request merged**